### DISCUSSION TITLE
Move Game files to External SSD

### DISCUSSION THREAD

**Discussion by dhrubanka** [PlayCover/PlayCover#473] [General]
I have a 256 GB version macbook air, since the space is very low I want to move the game files of Genshin Impact  to an external SSD like the rest of the applications that can be moved to SSD. However, in terms of PlayCover, there are multiple files in different directories related to Genshin Impact like containers and app files. Can this system really be executed from an external SSD?

**Comment by Candygoblen123** [comment]
Basically no.  While we could technically move the .app bundle off the boot drive, genshin impact will always look in its container folder for the downloaded assets. You can’t move that to any other drive.

**Reply by dhrubanka** [reply]
Maybe we can create symlinks for those files to an external SSD?

**Reply by dhrubanka** [reply]
https://www.imore.com/how-use-symbolic-links-macos-load-balance-your-data

**Reply by Candygoblen123** [reply]
I haven’t confirmed this myself, but I think I remember someone trying that and genshin just crashes. Feel free to try it yourself though.

**Reply by boa-z** [reply]
I have tried symbolic links. Now I can import ipa, but it doesn't show in playcover. Have you successfully done this?

```text
(base) boa@HX-2 Containers % pwd
/Users/boa/Library/Containers
(base) boa@HX-2 Containers % ls -al | grep playcover
lrwxr-xr-x@   1 boa  staff     61 Sep 28 22:55 io.playcover.PlayCover -> /Volumes/PC711/Applications/Containers/io.playcover.PlayCover
```

**Comment by ohaiibuzzle** [comment]
WIth the current system, would be very likely that you straight up cannot. By default a macOS application can only read:
>    At the specific direction of the user (so, TCC)
>    By using entitlements for specific file-system locations (described in Entitlements and System Resource Access)
>    When the file system location is in certain directories that are world readable

when sandboxed by the OS.

If it was really possible, it would be a serious security flaw (as an app can just make a symlink and reads anything it wants)

**Reply by ohaiibuzzle** [reply]
@ttuleyb You really are gonna do that and then not even at all realizes that will break aspects like App Preferences (as some of them are actually stored inside /Library of the Sandbox), which the app and OS (because App Sandbox rules) won't be able to read right?

Example:
<img width="1049" alt="image" src="https://github.com/PlayCover/PlayCover/assets/23693150/37ed1fb7-9457-4934-907f-54a61e262874">
